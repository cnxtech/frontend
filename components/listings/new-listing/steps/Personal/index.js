import React, { Component } from 'react'
import { Formik, Field } from 'formik'

import Input from '@emcasa/ui-dom/components/Input'
import Row from '@emcasa/ui-dom/components/Row'
import Col from '@emcasa/ui-dom/components/Col'
import View from '@emcasa/ui-dom/components/View'
import Text from '@emcasa/ui-dom/components/Text'
import NavButtons from 'components/listings/new-listing/shared/NavButtons'

class Personal extends Component {
  constructor(props) {
    super(props)
    this.nextStep = this.nextStep.bind(this)
    this.previousStep = this.previousStep.bind(this)
    this.validateName = this.validateName.bind(this)
    this.validateEmail = this.validateEmail.bind(this)
    this.updateStateFromProps = this.updateStateFromProps.bind(this)
    this.nameField = React.createRef()
  }

  state = {
    name: null,
    email: null
  }

  componentDidMount() {
    this.updateStateFromProps(this.props)
    this.nameField.current.focus()
  }

  componentWillReceiveProps(props) {
    this.updateStateFromProps(props)
  }

  updateStateFromProps(props) {
    const { personal } = props
    if (personal) {
      this.setState({
        name: personal.name,
        email: personal.email
      })
    }
  }

  nextStep() {
    const { navigateTo, updatePersonal } = this.props
    updatePersonal(this.state)
  }

  previousStep() {
    const { navigateTo } = this.props
    navigateTo('phone')
  }

  validateName(value) {
    if (!value) {
      return "Informe seu nome."
    }
  }

  validateEmail(value) {
    if (!value) {
      // It's ok if the user doesn't inform an e-mail
      return null
    }
    const regex = /\S+@\S+\.\S+/
    if (!value.match(regex)) {
      // But if they do, it has to be a valid one
      return "Informe um e-mail válido."
    }
  }

  render() {
    const { personal } = this.props
    let name, email
    if (personal) {
      name = personal.name
      email = personal.email
    }
    return (
      <div ref={this.props.hostRef}>
        <Row justifyContent="center">
          <Col width={[1, 1/2]}>
            <Formik
              initialValues={{
                name: name,
                email: email
              }}
              isInitialValid={() => {
                return !(this.validateName(name) || this.validateEmail(email))
              }}
              render={({isValid, setFieldTouched, setFieldValue, errors}) => (
                <>
                  <View body p={4}>
                    <Text
                      fontSize="large"
                      fontWeight="bold"
                      textAlign="center">
                      Qual seu nome e e-mail?
                    </Text>
                    <Text color="grey">Não se preocupe, não vamos encher sua caixa de spam.</Text>
                    <Row>
                      <Col width={1} mr={4}>
                        <Field
                          name="name"
                          validate={this.validateName}
                          render={({form}) => (
                            <Input
                              hideLabelView
                              ref={this.nameField}
                              placeholder="Nome*"
                              error={form.touched.name ? errors.name : null}
                              defaultValue={name}
                              onChange={(e) => {
                                const { value } = e.target
                                setFieldValue('name', value)
                                setFieldTouched('name')
                                this.setState({name: value})
                              }}
                            />
                          )}/>
                      </Col>
                    </Row>
                    <Row>
                      <Col width={1} mr={4}>
                        <Field
                          name="email"
                          validate={this.validateEmail}
                          render={({form}) => (
                            <Input
                              hideLabelView
                              placeholder="E-mail"
                              error={form.touched.email ? errors.email : null}
                              defaultValue={email}
                              onChange={(e) => {
                                const { value } = e.target
                                setFieldValue('email', value)
                                setFieldTouched('email')
                                this.setState({email: value})
                              }}
                            />
                          )}/>
                      </Col>
                    </Row>
                  </View>
                  <View bottom p={4}>
                    <NavButtons
                      previousStep={this.previousStep}
                      nextStep={this.nextStep}
                      nextEnabled={isValid}
                    />
                  </View>
                </>
              )}
            />
          </Col>
        </Row>
      </div>
    )
  }
}

export default Personal
