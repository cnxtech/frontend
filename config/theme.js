import theme from '@emcasa/ui'
import {fas} from '@fortawesome/pro-solid-svg-icons'
import {fal} from '@fortawesome/pro-light-svg-icons'
import icons from './icons'

export default {
  ...theme,
  icons: {emcasa: icons, default: fas, light: fal}
}
