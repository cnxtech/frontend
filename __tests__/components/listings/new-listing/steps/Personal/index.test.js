import React from 'react'
import Personal from 'components/listings/new-listing/steps/Personal'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import theme from '@emcasa/ui'
import { ThemeProvider } from 'styled-components'

it('renders correctly', () => {
  const tree = renderer.create(
    <ThemeProvider theme={theme}>
      <Personal />
    </ThemeProvider>).toJSON()
  expect(tree).toMatchSnapshot()
})
