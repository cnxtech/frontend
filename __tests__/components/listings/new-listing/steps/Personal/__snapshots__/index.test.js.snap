// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c2 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: #3c4858;
  font-size: 24px;
  font-weight: bold;
  text-align: center;
}

.c3 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: #979797;
  font-size: 18px;
}

.c9 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: #3c4858;
  font-size: 14px;
}

.c10 {
  -webkit-font-smoothing: antialiased;
  font-family: FaktSoftPro-Normal;
  color: #ff4949;
  font-size: 18px;
}

.c6 {
  -webkit-font-smoothing: antialiased;
  display: block;
  box-sizing: border-box;
  width: 100%;
  outline: none;
  border-radius: 4px;
  border: 1px solid;
  border-color: #e0e6ed;
  padding: 0px 10px 0px 10px;
  height: 60px;
  color: #3c4858;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
}

.c6:focus {
  border-radius: 4px;
  border: 1px solid;
  border-color: #1fb6ff;
  padding: 0px 10px 0px 10px;
  height: 60px;
}

.c6::-webkit-input-placeholder {
  color: #8492a6;
}

.c6::-moz-placeholder {
  color: #8492a6;
}

.c6:-ms-input-placeholder {
  color: #8492a6;
}

.c6::placeholder {
  color: #8492a6;
}

.c7 {
  height: 25px;
}

.c8 {
  display: inline;
  color: #ff4949;
  font-family: FaktSoftPro-Normal;
  font-size: 14px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 20px;
  padding-top: 0px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 20px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  width: 100%;
}

.c5 {
  margin-right: 20px;
  width: 100%;
}

.c12 {
  width: 41.66666666666667%;
}

.c13 {
  outline: none;
  display: block;
  cursor: pointer;
  padding: 0 20px 0 20px;
  border-radius: 4px;
  border: 1px solid;
  border-color: #e0e6ed;
  background-color: #ffffff;
  width: 100%;
  height: 60px;
  color: #3c4858;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
  border-color: #e0e6ed;
}

.c13 :hover {
  background-color: #ffffff;
}

.c14 {
  outline: none;
  display: block;
  cursor: default;
  padding: 0 20px 0 20px;
  border-radius: 4px;
  border: 1px solid;
  border-color: #e0e6ed;
  background-color: #ffffff;
  width: 100%;
  height: 60px;
  color: #8492a6;
  font-family: FaktSoftPro-Normal;
  font-size: 18px;
  -webkit-font-smoothing: antialiased;
  border-color: #e0e6ed;
}

.c14 :hover {
  background-color: #ffffff;
}

@media screen and (min-width:780px) {
  .c1 {
    width: 50%;
  }
}

<div>
  <div
    className="c0"
  >
    <div
      className="c1"
      width={
        Array [
          1,
          0.5,
        ]
      }
    >
      <p
        className="c2"
        color="dark"
        fontSize="large"
        fontWeight="bold"
      >
        Qual seu nome e e-mail?
      </p>
      <p
        className="c3"
        color="grey"
        fontSize="default"
      >
        Não se preocupe, não vamos encher sua caixa de spam.
      </p>
      <div
        className="c4"
      >
        <div
          className="c5"
          width={1}
        >
          <div
            className=""
          >
            <input
              className="c6"
              fontSize="default"
              height="tall"
              letterSpacing="default"
              onChange={[Function]}
              placeholder="Nome*"
            />
            <div
              className="c7 "
            >
              <span
                className="c8 c9"
                color="dark"
                fontSize="small"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="c4"
      >
        <div
          className="c5"
          width={1}
        >
          <div
            className=""
          >
            <input
              className="c6"
              fontSize="default"
              height="tall"
              letterSpacing="default"
              onChange={[Function]}
              placeholder="E-mail"
            />
            <div
              className="c7 "
            >
              <span
                className="c8 c9"
                color="dark"
                fontSize="small"
              />
            </div>
          </div>
        </div>
      </div>
      <p
        className="c10"
        color="red"
        fontSize="default"
      />
      <div
        className="c11"
      >
        <div
          className="c12"
          width={0.4166666666666667}
        >
          <button
            className="c13"
            fontSize="default"
            height="tall"
            onClick={[Function]}
          >
            Voltar
          </button>
        </div>
        <div
          className="c12"
          width={0.4166666666666667}
        >
          <button
            className="c14"
            disabled={true}
            fontSize="default"
            height="tall"
            onClick={[Function]}
          >
            Avançar
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
